src/CBT.NuGet/Internal/NuGetPropertyGenerator.cs

```diff
@@ -1,68 +1,80 @@
+        public bool Generate(string outputPath, string propertyVersionNamePrefix, string propertyPathNamePrefix, PackageRestoreData restoreData)
         {
             ProjectRootElement project = ProjectRootElement.Create();
             ProjectPropertyGroupElement propertyGroup = project.AddPropertyGroup();
             propertyGroup.SetProperty("MSBuildAllProjects", "$(MSBuildAllProjects);$(MSBuildThisFileFullPath)");
 
             ProjectItemGroupElement itemGroup = project.AddItemGroup();
-            foreach (PackageIdentityWithPath packageInfo in ParsePackages(nuGetPackagesPath, restoreData))
+
+            foreach (string packageConfigPath in _packageConfigPaths)
             {
-                propertyGroup.SetProperty(
-                    String.Format(CultureInfo.CurrentCulture, "{0}{1}", propertyPathNamePrefix,
-                        packageInfo.Id.Replace(".", "_")),
-                    String.Format(CultureInfo.CurrentCulture, "{0}", packageInfo.FullPath));
-                propertyGroup.SetProperty(
-                    String.Format(CultureInfo.CurrentCulture, "{0}{1}", propertyVersionNamePrefix,
-                        packageInfo.Id.Replace(".", "_")),
-                    String.Format(CultureInfo.CurrentCulture, "{0}", packageInfo.Version.ToString()));
-                // Consider adding item metadata of packageid and version for ease of consumption of this property.
-                itemGroup.AddItem("CBTNuGetPackageDir",
-                    String.Format(CultureInfo.CurrentCulture, "{0}", packageInfo.FullPath));
+                _logger.LogMessage(MessageImportance.Low, $"Parsing '{packageConfigPath}'");
+
+                IEnumerable<PackageIdentityWithPath> parsedPackages = null;
+
+                INuGetPackageConfigParser configParser = _configParsersLazy.Value.FirstOrDefault(i => i.TryGetPackages(packageConfigPath, restoreData, out parsedPackages));
+
+                if (configParser != null && parsedPackages != null)
+                {
+                    foreach (PackageIdentityWithPath packageInfo in parsedPackages)
+                    {
+                        propertyGroup.SetProperty($"{propertyPathNamePrefix}{packageInfo.Id.Replace(".", "_")}", $"{packageInfo.FullPath}");
+
+                        propertyGroup.SetProperty($"{propertyVersionNamePrefix}{packageInfo.Id.Replace(".", "_")}", $"{packageInfo.Version.ToString()}");
+
+                        // Consider adding item metadata of packageid and version for ease of consumption of this property.
+                        itemGroup.AddItem("CBTNuGetPackageDir", packageInfo.FullPath);
+                    }
+                }
             }
-            project.Save(outputPath);
 
-            return true;
-        }
+            if (propertyGroup.Properties.Count > 1)
```

Comment:
>1 [](start = 49, length = 1)

add comment on why 1

Reply:
I introduced a variable to make it more apparent